src/net/neighbors.rs

```diff
@@ -1890,10 +1890,16 @@ impl PeerNetwork {
         Ok((count, num_allowed_peers as u64))
     }
 
-    /// Instantiate the neighbor walk to an always-allowed node
-    fn instantiate_walk_to_always_allowed(&mut self) -> Result<(), net_error> {
-        let allowed_peers =
-            PeerDB::get_always_allowed_peers(self.peerdb.conn(), self.local_peer.network_id)?;
+    /// Instantiate the neighbor walk to an always-allowed node.
+    /// If we're in the initial block download, then this must also be a *bootstrap* peer.
+    fn instantiate_walk_to_always_allowed(&mut self, ibd: bool) -> Result<(), net_error> {
+        let allowed_peers = if ibd {
+            // only get bootstrap peers
+            PeerDB::get_bootstrap_peers(&self.peerdb.conn(), self.local_peer.network_id)?
```

Comment:
Do we want to return the subset of bootstrap peers that are allowed?

Reply:
A peer can be a bootstrap peer without being allowed.  Specifically, if I put a node in the `bootstrap_node` list in my config file, it should override any other decision the node software makes about whether or not that peer should otherwise be ignored.

Reply:
A node can be either "allowed" or "denied" for an interval of time between now, and a future timestamp _t_.  This is how the node temporarily bans misbehaving peers, for example -- it marks them as "denied" until _t_, where _t_ grows exponentially up to a cap.  However, a node can also be marked as "always-allowed", which means "allowed, and will never be denied".  This includes bootstrap nodes.

In addition, a node can be marked as "initial", in which case, it will always be considered a candidate for inventory synchronization during initial block download.  Once the node reaches the chain tip, this consideration stops.